sa/sa.go

```diff
@@ -111,19 +111,21 @@ func updateChallenges(authID string, challenges []core.Challenge, tx *gorp.Trans
 
 // GetRegistration obtains a Registration by ID
 func (ssa *SQLStorageAuthority) GetRegistration(ctx context.Context, id int64) (core.Registration, error) {
-	regObj, err := ssa.dbMap.Get(regModel{}, id)
+	var reg regModel
+	err := safeSelectOne(
+		ssa.dbMap.SelectOne,
+		&reg,
+		fmt.Sprintf("SELECT %s FROM registrations WHERE id = %d", regFields, id),
+	)
 	if err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
This doesn't match the pattern below, which checks in this order:

```
if err == sql.ErrNoRows {
...
if err != nil {
```

Probably better to match the existing pattern for now, and refactor separately if you think the nested format here is preferable.